### ISSUE TITLE
How to make full use of CPU?

### ISSUE THREAD

**Issue by EquatorSavage** [UKPLab/sentence-transformers#592]
**`from sentence_transformers import SentenceTransformer`
`model_fpath = "./distiluse-base-multilingual-cased"`
`model = SentenceTransformer(model_fdir,device="cpu")`
`model.encode("I am a",convert_to_tensor=True,num_workers=32)`**

When I do high concurrency testing with locust.
I monitor the server, the load of CPU is low, only about 30%.

**Comment by nreimers** [comment]
Num workers only controls the number of threads for tokenization. It is best to leave it at 0.

Have a look at the multi process encode method when you have to encode large sets of sentences

**Comment by EquatorSavage** [comment]
@nreimers 
Thanks for your reply.
My program is multi process.
I tried set num_workers 0 and it worked,the load of CPU is 30%~50%.
I have also tried google's model universal-sentence-encoder-multilingual-large_1,the load of CPU is very high,about 80%~90%.
When I increase concurrency,Google's model encoding time is very stable, no change,but distiluse-base-multilingual-cased's encoding time has grown a lot longer.
